codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/waiters/KotlinJmespathExpressionVisitor.kt

```diff
@@ -226,6 +226,39 @@ class KotlinJmespathExpressionVisitor(
             VisitedExpression(ident)
         }
 
+        "abs" -> {
+            codegenReq(expression.arguments.size == 1) { "Unexpected number of arguments to $expression" }
+
+            val number = acceptSubexpression(expression.arguments[0])
+
+            val absExpr = ensureNullGuard(number.shape, "let { kotlin.math.abs(it.toDouble()) }")
+            val ident = addTempVar("abs", "${number.identifier}$absExpr")
+
+            VisitedExpression(ident, number.shape)
+        }
+
+        "floor" -> {
+            codegenReq(expression.arguments.size == 1) { "Unexpected number of arguments to $expression" }
+
+            val number = acceptSubexpression(expression.arguments[0])
+
+            val absExpr = ensureNullGuard(number.shape, "let { kotlin.math.floor(it.toDouble()) }")
+            val ident = addTempVar("floor", "${number.identifier}$absExpr")
+
+            VisitedExpression(ident, number.shape)
+        }
+
+        "ceil" -> {
+            codegenReq(expression.arguments.size == 1) { "Unexpected number of arguments to $expression" }
+
+            val number = acceptSubexpression(expression.arguments[0])
+
+            val absExpr = ensureNullGuard(number.shape, "let { kotlin.math.ceil(it.toDouble()) }")
+            val ident = addTempVar("ceil", "${number.identifier}$absExpr")
+
+            VisitedExpression(ident, number.shape)
+        }
```

Comment:
**Nit**: By happy coincidence, the JMESPath names for these functions are the same in Kotlin. We can take advantage of that to collapse these three branches into one:

```kotlin
override fun visitFunction(expression: FunctionExpression): VisitedExpression = when (expression.name) {
    "contains" -> ...
    "length" -> ...

    "abs", "floor", "ceil" -> {
        codegenReq(expression.arguments.size == 1) { "Unexpected number of arguments to $expression" }

        val number = acceptSubexpression(expression.arguments[0])

        val expr = ensureNullGuard(number.shape, "let { kotlin.math.${expression.name}(it.toDouble()) }")
        val ident = addTempVar(expression.name, "${number.identifier}$expr")

        VisitedExpression(ident, number.shape)
    }
}
```